transcom/mymove
# 19782 shipment cancellation date validation be main

## [B-19782](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A946146)

## Summary

Integration PR: https://github.com/transcom/mymove/pull/12931
Integration PR Kickback: https://github.com/transcom/mymove/pull/12979

This PR prevents the TOO from cancelling any shipment that's 'actualPickupdate' is on or after the request date.
If there is not actualPickupdate set then the shipment's cancellation will go through.

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test with actualPickupdate

1. Make a HHG 
2. Login as SC to edit and submit move 
3. Login as Prime to update shipment with an 'actualPickupDate' on or after today
4. Login as TOO and try to cancel the move
5. Confirm the Update error message

### How to test without actualPickupdate

1. Make a HHG
2. Login as SC to edit and submit move
3. Login as TOO and approve and send move
4. Go to the 'Move Task Order' tab and try to cancel the move
5. Confirm you're able to delete the move without an actualPickupdate

### Backend


## Screenshots

**_With invalid actualPickupdate_**
<img width="1145" alt="Screenshot 2024-06-05 at 1 23 44 PM" src="https://github.com/transcom/mymove/assets/146969726/170aa7b2-aa9b-4614-a19e-34d2780cb8f6">

**_with valid or no actualPickupdate_**
<img width="1049" alt="Screenshot 2024-06-07 at 10 23 00 AM" src="https://github.com/transcom/mymove/assets/146969726/6ab95c7b-f65e-46a1-90f2-4e6344ade237">
